pipeline/eBOSS_QSO/QuasarMapper.py

```diff
@@ -0,0 +1,191 @@
+import matplotlib.pyplot as plt
+from astropy.io import fits
+from astropy.table import Table
+
+import os
+import pandas as pd
+import collections
+import numpy as np
+import healpy as hp
+
+import pymaster as nmt
+
+
+class QuasarMapper:
+    def __init__(self): 
+        #Inputs: the four fits files from eBOSS DR14
+        
+        data_path = '/home/zcapjru/PhD/Data/'
+        #Check for data files:
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_data-NGC-vDR16.fits') :
+            print('NGC data found')
+            self.NGC_data =  fits.open(data_path + 'eBOSS_QSO_clustering_data-NGC-vDR16.fits')
+        else :
+            print('missing NGC data')
+
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_data-SGC-vDR16.fits') :
+            print('SGC data found')
+            self.SGC_data =  fits.open(data_path + 'eBOSS_QSO_clustering_data-SGC-vDR16.fits')
+        else :
+            print('missing SGC data')
+           
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_random-NGC-vDR16.fits') :
+            print('NGC random data found')
+            self.NGC_r_data =  fits.open(data_path + 'eBOSS_QSO_clustering_random-NGC-vDR16.fits')
+        else :
+            print('missing NGC random data')
+           
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_random-SGC-vDR16.fits') :
+            print('SGC random data found')
+            self.SGC_r_data =  fits.open(data_path + 'eBOSS_QSO_clustering_random-SGC-vDR16.fits')
+        else :
+            print('missing SGC random data')
+        
+
+        self.nside = 512
+        #set resolution
+        
+        self.bands = nmt.NmtBin.from_nside_linear(self.nside, 20)
+        #set l's per band
+        
+        self.NGC = Table.read(self.NGC_data).to_pandas()
+        self.NGC_r = Table.read(self.NGC_r_data).to_pandas()
+        self.SGC = Table.read(self.SGC_data).to_pandas()
+        self.SGC_r = Table.read(self.SGC_r_data).to_pandas()
+        #from .fits to table to managable pandas frames
+
+        #Merge NGC + SGC
+        self.edges = [1.5] #inner bins edges
+        self.whole =  pd.concat([self.NGC, self.SGC])

```

Comment:
Here you could do `self.whole = pd.concat(self.cat_data)`. Doing so you will be storing things twice, so you could even overwrite `cat_data` (i.e. `self.cat_data = pd.concat(self.cat_data)`).

Do the same with `self.whole_r` below.